Android/Diwakar Prajapati/Newsopedia/app/src/main/java/com/example/newsopedia/ui/technology/TechnologyFragment.java

```diff
@@ -0,0 +1,86 @@
+package com.example.newsopedia.ui.technology;
+
+import android.os.Bundle;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.EditText;
+import android.widget.TextView;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.fragment.app.Fragment;
+import androidx.lifecycle.Observer;
+import androidx.lifecycle.ViewModelProvider;
+import androidx.navigation.ui.AppBarConfiguration;
+import androidx.recyclerview.widget.LinearLayoutManager;
+import androidx.recyclerview.widget.RecyclerView;
+
+import com.example.newsopedia.Adapter;
+import com.example.newsopedia.ApiClient;
+import com.example.newsopedia.Model.Articles;
+import com.example.newsopedia.Model.Category;
+import com.example.newsopedia.R;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
+
+public class TechnologyFragment extends Fragment {
+
+    private TechnologyViewModel slideshowViewModel;
+
+    private AppBarConfiguration mAppBarConfiguration;
+    final String API_KEY = "58a952499a7341b3955c30e0798bbb42";
+    RecyclerView recyclerView;
+    Adapter adapter;
+    List<Articles> articles = new ArrayList<>();
+
+    public View onCreateView(@NonNull LayoutInflater inflater,
+                             ViewGroup container, Bundle savedInstanceState) {
+        slideshowViewModel =
+                new ViewModelProvider(this).get(TechnologyViewModel.class);
+        View root = inflater.inflate(R.layout.fragment_home, container, false);
+
+        recyclerView = root.findViewById(R.id.recyclerView);
+        recyclerView.setLayoutManager(new LinearLayoutManager(getContext()));
+        retrieveJSON("technology",API_KEY);
+
+        EditText editText = root.findViewById(R.id.editText);
+        root.findViewById(R.id.button).setOnClickListener(new View.OnClickListener() {

```

Comment:
Anonymous listeners can be replaced with lambda. Please try.

Reply:
Done
But at one place writing directly using onClick call back seemed better so, because if I use lambda function then I will have to pass multiple arguments to the function.